### PR TITLE
Linking t8code_jll against p4est_jll

### PR DESCRIPTION
**Pull request by jmark**

### CODE REVIEW

File: T/t8code/build_tarballs.jl [hunk 1]
```diff
@@ -52,7 +52,7 @@ else
 fi
 
 # Run configure
-./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-static --without-blas ${mpiopts}
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} --disable-static --without-blas ${mpiopts}  --with-sc=/workspace/destdir --with-p4est=/workspace/destdir
```
[the remark this answers is not in the document]
Comment (jmark) [reply T/t8code/build_tarballs.jl]: Done.

**Comment by jmark** [comment]
This t8code wrapper now depends on `p4est_jll` and successfully compiles on all tested platforms.

@sloede @giordano @staticfloat Please review and merge.

**Review by sloede — APPROVED** [review]
Nice work, LGTM!

**Comment by giordano** [comment]
Please, do ***NOT*** rebase/merge on master, the caching system we have in place makes it harder than necessary to retrieve the build logs for inspection, as they'll be buried deep down in previous builds.

File: T/t8code/build_tarballs.jl [hunk 2]
```diff
@@ -11,14 +11,15 @@ version = v"1.1.2"
...
+                  "8a30206a8fb47013b3dafe7565cf8e09023df8373c1049e7e231d9fd36b011e4"),
+
     DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
-cd t8code/
+cd t8code
```
Comment (giordano) [inline T/t8code/build_tarballs.jl:22]:
```
cd $WORKSPACE/srcdir/t8code
```

File: T/t8code/build_tarballs.jl [hunk 3]
```diff
@@ -11,14 +11,15 @@ version = v"1.1.2"
 # Collection of sources required to complete build
 sources = [
     ArchiveSource("https://github.com/DLR-AMR/t8code/releases/download/v$(version)/t8code_v$(version).tar.gz",
-                  "0bd4bee6694735d14fb4274275fb8c4bdeacdbd29b257220c308be63e98be8f7"),
+                  "8a30206a8fb47013b3dafe7565cf8e09023df8373c1049e7e231d9fd36b011e4"),
```
Comment (giordano) [inline T/t8code/build_tarballs.jl:14]: Uhm, the tag is now pointing to a different commit?

Comment (jmark) [reply T/t8code/build_tarballs.jl:14]: Yes, I updated the t8code tarball. This points to a pre-release version of t8code which will be released in the near to midterm future. I cannot wait for that and need an official jll package as soon as possible.

File: T/t8code/build_tarballs.jl (hunk 2 above)
Comment (jmark) [reply T/t8code/build_tarballs.jl]: Done.

**Comment by jmark** [comment]
Now all checks successfully passed.

@giordano Is there anything more I should improve before merging?

**Pull request merged**